gpt4all-bindings/python/gpt4all/pyllmodel.py

```diff
@@ -245,35 +241,38 @@ def prompt_model(
         repeat_last_n: int = 10,
         context_erase: float = 0.75,
         reset_context: bool = False,
-        streaming=False,
-    ) -> str:
+    ):
         """
         Generate response from model from a prompt.
 
         Parameters
         ----------
         prompt: str
             Question, task, or conversation for model to respond to
-        streaming: bool
-            Stream response to stdout
+        callback(token_id:int, response:str): bool
+            The model sends response tokens to callback
 
         Returns
         -------
-        Model response str
+        None
         """
 
-        prompt_bytes = prompt.encode('utf-8')
-        prompt_ptr = ctypes.c_char_p(prompt_bytes)
-
-        old_stdout = sys.stdout
+        logging.info("pyllmodel.py: LLModel.prompt_model -- prompt:\n" +
+                     "%s\n" +
+                     "===/pyllmodel.py: LLModel.prompt_model -- prompt/===", prompt)
 
-        stream_processor = DualStreamProcessor()
 
-        if streaming:
-            stream_processor.stream = sys.stdout
+        prompt_bytes = prompt.encode('utf-8')
+        prompt_ptr = ctypes.c_char_p(prompt_bytes)
 
-        sys.stdout = stream_processor
+        def _callback_wrapper(callback: ResponseCallbackType) -> ResponseCallbackType:
+            
+            def _callback(token_id: int, response: str) -> bool:
+                nonlocal callback
+                return callback(token_id, response.decode('utf-8', 'replace'))
 
+            return _callback
```

Comment:
And here instead of:
```py
        def _callback_wrapper(callback: ResponseCallbackType) -> ResponseCallbackType:
            
            def _callback(token_id: int, response: str) -> bool:
                nonlocal callback
                return callback(token_id, response.decode('utf-8', 'replace'))

            return _callback
```
it should be:
```py
        def _callback_wrapper(callback: ResponseCallbackType) -> RawResponseCallbackType:
            
            def _callback(token_id: int, response: bytes) -> bool:
                nonlocal callback
                return callback(token_id, response.decode('utf-8', 'replace'))

            return _callback
```
Then as explained:
- please don't nest it inside `prompt_model()`.
- better naming would be neat.